### DISCUSSION TITLE
Object display name / Plugin Support for Existing Models

### DISCUSSION THREAD

**Discussion by mrkeathley** [netbox-community/netbox#10462] [Q&A]
I have been testing Netbox for a few weeks and really like it. I have added a plugin to handle a number of my specific project requirements. 

However, I am at a hurdle with needing to change the display name to a couple of models. We use Customer instead of Tenant. And I am using Site to simply be our list of Buildings.  Is there a way to update the meta on an existing model? I could set the verbose_name field and get halfway there. Is there a better way to do this? 

We also want to navigate to a building or customer and have an auto-prefix assignment (select the subnet size and auto grab the next available block of sufficient size). I can do this in a plugin stand alone page but would like to update the existing view. 

Am I missing a guide here or am I off the normal trail?

**Comment by mtinberg** [comment]
I think if you want to customize behavior this way either a Netbox Script (simple one-page form with custom Python to implement your own logic) or a Plugin (for a more complicated form, or multi-step process) that layers on top of Netbox cleanly is better for long-term maintainability than trying to modify the base system, and then have to merge/rebase your own fork for the rest of time. You still have to test your upgrades for breaking changes but that should be simpler with a Script/Plugin.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Matthew Keathley ***@***.***>
Sent: Monday, September 26, 2022 8:59 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Object display name / Plugin Support for Existing Models (Discussion #10462)


I have been testing Netbox for a few weeks and really like it. I have added a plugin to handle a number of my specific project requirements.

However, I am at a hurdle with needing to change the display name to a couple of models. We use Customer instead of Tenant. And I am using Site to simply be our list of Buildings. Is there a way to update the meta on an existing model? I could set the verbose_name field and get halfway there. Is there a better way to do this?

We also want to navigate to a building or customer and have an auto-prefix assignment (select the subnet size and auto grab the next available block of sufficient size). I can do this in a plugin stand alone page but would like to update the existing view.

Am I missing a guide here or am I off the normal trail?
